src/EnergyPlus/HVACUnitaryBypassVAV.cc

```diff
@@ -3114,147 +2990,146 @@ namespace HVACUnitaryBypassVAV {
-                } else if (CBVAV(CBVAVNum).DXCoolCoilType_Num == DataHVACGlobals::CoilDX_CoolingSingleSpeed) {
+                    state.dataHVACUnitaryBypassVAV->SaveCompressorPLR = state.dataDXCoils->DXCoilPartLoadRatio(cBVAV.DXCoolCoilIndexNum);
+                } else if (cBVAV.DXCoolCoilType_Num == DataHVACGlobals::CoilDX_CoolingSingleSpeed) {
                     DXCoils::SimDXCoil(state,
-                                       CBVAV(CBVAVNum).DXCoolCoilName,
+                                       cBVAV.DXCoolCoilName,
                                        DataHVACGlobals::CompressorOperation::Off,
                                        FirstHVACIteration,
-                                       CBVAV(CBVAVNum).CoolCoilCompIndex,
+                                       cBVAV.CoolCoilCompIndex,
                                        DataHVACGlobals::ContFanCycCoil,
                                        0.0,
                                        OnOffAirFlowRatio);
-                    state.dataHVACUnitaryBypassVAV->SaveCompressorPLR = state.dataDXCoils->DXCoilPartLoadRatio(CBVAV(CBVAVNum).DXCoolCoilIndexNum);
-                } else if (CBVAV(CBVAVNum).DXCoolCoilType_Num == DataHVACGlobals::CoilDX_CoolingTwoStageWHumControl) {
+                    state.dataHVACUnitaryBypassVAV->SaveCompressorPLR = state.dataDXCoils->DXCoilPartLoadRatio(cBVAV.DXCoolCoilIndexNum);
+                } else if (cBVAV.DXCoolCoilType_Num == DataHVACGlobals::CoilDX_CoolingTwoStageWHumControl) {
                     DXCoils::SimDXCoilMultiMode(state,
-                                                CBVAV(CBVAVNum).DXCoolCoilName,
+                                                cBVAV.DXCoolCoilName,
                                                 DataHVACGlobals::CompressorOperation::Off,
                                                 FirstHVACIteration,
                                                 0.0,
                                                 0,
-                                                CBVAV(CBVAVNum).CoolCoilCompIndex,
+                                                cBVAV.CoolCoilCompIndex,
                                                 DataHVACGlobals::ContFanCycCoil);
-                    state.dataHVACUnitaryBypassVAV->SaveCompressorPLR = state.dataDXCoils->DXCoilPartLoadRatio(CBVAV(CBVAVNum).DXCoolCoilIndexNum);
-                } else if (CBVAV(CBVAVNum).DXCoolCoilType_Num == DataHVACGlobals::Coil_CoolingAirToAirVariableSpeed) {
+                    state.dataHVACUnitaryBypassVAV->SaveCompressorPLR = state.dataDXCoils->DXCoilPartLoadRatio(cBVAV.DXCoolCoilIndexNum);
+                } else if (cBVAV.DXCoolCoilType_Num == DataHVACGlobals::Coil_CoolingAirToAirVariableSpeed) {
                     Real64 QZnReq(0.0);                // Zone load (W), input to variable-speed DX coil
```

Comment:
Look mom, I'm calling the Real64 constructor!

Reply:
[[90297ce](https://github.com/NREL/EnergyPlus/pull/10018/commits/90297ce757aad7cd7d65963604c1b71eff63b914)](https://github.com/NREL/EnergyPlus/pull/10018/commits/90297ce757aad7cd7d65963604c1b71eff63b914)